src/doc/man/cargo-tree.adoc

```diff
@@ -0,0 +1,193 @@
+= cargo-tree(1)
+:idprefix: cargo_tree_
+:doctype: manpage
+:actionverb: Display
+:noall: true
+
+== NAME
+
+cargo-tree - Display a tree visualization of a dependency graph
+
+== SYNOPSIS
+
+`cargo tree [_OPTIONS_]`
+
+== DESCRIPTION
+
+This command will display a tree of dependencies to the terminal. An example of a simple project that depends on the "rand" package:
+
+----
+myproject v0.1.0 (/myproject)
+└── rand v0.7.3
+    ├── getrandom v0.1.14
+    │   ├── cfg-if v0.1.10
+    │   └── libc v0.2.68
+    ├── libc v0.2.68 (*)
+    ├── rand_chacha v0.2.2
+    │   ├── ppv-lite86 v0.2.6
+    │   └── rand_core v0.5.1
+    │       └── getrandom v0.1.14 (*)
+    └── rand_core v0.5.1 (*)
+[build-dependencies]
+└── cc v1.0.50
+----
+
+Packages marked with `(*)` have been "de-duplicated". The dependencies for the
+package have already been shown elswhere in the graph, and so are not
+repeated. Use the `--no-dedupe` option to repeat the duplicates.
+
+== OPTIONS
+
+=== Tree Options
+
+*-i*::
+*--invert*::
+    Invert the tree. Typically this flag is used with the `-p` flag to show
+    the dependencies for a specific package.
+
+*--no-dedupe*::
```

Comment:
Over time I've tended to prefer "positive" CLI flags to negative ones. The main reason is more rustc-specific when these flags take values, but I find double-negatives really hard to read (`-C no-foo=no` means "foo=yes" actually). Do we want to consider renaming these flags so it's something like `--deduplicate=no` or `--dev-dependencies=no` perhaps?

Reply:
I agree that avoiding double negatives is good, but in this case I don't think it is important. There isn't a corresponding flag for `dedupe` since that is the default.  I don't feel like this needs to be super flexible, so I don't think it should take a value (and swapping the "no" from the front to the back doesn't substantially change things).

I do agree the flag itself is potentially confusing.  Other alternatives I thought of was `--show-dupes` or `--show-duplicates`, but it's starting to get long and wordy.  I was also concerned about confusion with the `--duplicates` flag, which means something completely different.

One (tangentially related) idea I wanted to toss out is changing `(*)` to `deduped`.  This is what NPM uses, so there might be some familiarity there.  With the explicit wording, it might make more sense?  I mention this because I remember the first time I used `cargo tree` I was confused what the `(*)` signified.

Any other ideas?

Reply:
Oh right I forgot about `--duplicates`. Given the other consolidations that have been happening, I agree let's leave this as `--no-dedupe`

I like the idea though of changing `(*)` to something else. Do you think it'd be useful to take that a bit further and do `(deduped from line 13)` so if you pipe the output to a text editor you can jump around? Or something that points to where the deduplication happened at least.

Reply:
Sounds interesting.  I'd like to put this on the future enhancements pile, though.